index.js

```diff
@@ -731,7 +731,32 @@ function flameGraph (opts) {
           .attr('transition', 'transform 200ms ease-in-out')
           .call(panZoom)
           .on('wheel.zoom', null)
-          .on('dblclick.zoom', null)
+          .on('contextmenu', function () {
+            const pointerCoords = { x: d3.event.offsetX, y: d3.event.offsetY }
+            const target = getNodeAt(this, pointerCoords.x, pointerCoords.y)
+
+            d3.event.preventDefault()
+
+            if (target) {
+              // Passes original datum and rect / event co-ordinates, same as hoverin / hoverout dispatches
+              dispatch.call('contextmenu', null, target.data, getNodeRect(target), pointerCoords)
+            } else {
+              // Click on the flamegraph background. Listeners can ignore it or treat it as deselection
+              dispatch.call('contextmenu', null, null, null, null)
+            }
+          })
+          .on('dblclick.zoom', function () {
```

Comment:
I'd prefer keeping the `.on('dblclick.zoom', null)` call above and just doing `.on('dblclick')` here, since the intention is to disable the d3-zoom behaviour (remove event from `.zoom` namespace) and instead use our own custom behaviour (add event without namespace).